stdlib/docs/crypto/hashes/rpo.md

```diff
@@ -10,5 +10,8 @@ Prepares the top of the stack with the hasher initial state.<br /><br />This pro
+| hash_memory_with_state | Computes hash of Felt values starting at the specified memory address using the provided hasher<br />state.<br /><br />This procedure divides the hashing process into two parts: hashing pairs of words using<br />`absorb_double_words_from_memory` procedure and hashing the remaining values using the `permute`<br />procedure.<br /><br />Inputs:  [C, B, A, ptr, end_pairs_addr, num_elements%8]<br />Outputs: [HASH]<br /><br />Where:<br />- ptr is the  memory address of the first element to be hashed. This address must be<br />word-aligned - i.e., divisible by 4.<br />- C, B, A are three words representing the hasher state.<br />- end_pairs_addr is the memory address at which the pairs of words end.<br />- num_elements%8 is the number of elements which didn't fit to the word pairs and should be<br />hashed separately.<br />- HASH is the resulting hash of the provided memory values.<br /> |
+| hash_elements | Computes hash of Felt values starting at the specified memory address.<br /><br />This procedure divides the hashing process into two parts: hashing pairs of words using<br />`absorb_double_words_from_memory` procedure and hashing the remaining values using the `permute`<br />procedure.<br /><br />Inputs:  [ptr, num_elements]<br />Outputs: [HASH]<br /><br />Where:<br />- ptr is the  memory address of the first element to be hashed. This address must be<br />word-aligned - i.e., divisible by 4.<br />- num_elements is the number of elements to be hashed.<br /><br />Cycles:<br />- If number of elements divides by 8: 52 cycles + 3 * words<br />- Else: 185 cycles + 3 * words<br />where `words` is the number of quads of input values.<br /> |
+| hash | Computes RPO hash of a single 256-bit input (1 word = 4 field elements).<br /><br />Input: [A, ...]<br />Output: [B, ...]<br /><br />Where:<br />- A = [a_0, a_1, a_2, a_3] represents 4 field elements (1 word)<br />- B = [b_0, b_1, b_2, b_3] = RPO256(A)<br /> |
+| merge | Merges two 256-bit digests (2 words = 8 field elements) via RPO hash.<br /><br />Input: [B, A, ...]<br />Output: [C, ...]<br /><br />Where:<br />- A = [a_0, a_1, a_2, a_3] represents first digest (1 word)<br />- B = [b_0, b_1, b_2, b_3] represents second digest (1 word)<br />- C = [c_0, c_1, c_2, c_3] = RPO256(A \|\| B)<br /> |
+| permute | Performs RPO permutation on the hasher state.<br /><br />Input: [C, B, A, ...]<br />Output: [C', B', A', ...]<br /><br />Where:<br />- A, B, C are the three words of the hasher state<br />- A', B', C' are the permuted state words<br /><br />Cycles: 1<br /> |
```

Comment:
Not sure what these doc changes are (and similarly in other files) - are these meant to be there?

Reply:
I'll double check but I got these after rebuilding with BUILDDOCS. I think next needs to be rebuilt.